src/globus_sdk/services/auth/client/service_client.py

```diff
@@ -631,3 +680,1187 @@ def delete_project(self, project_id: UUIDLike) -> GlobusHTTPResponse:
+        :type high_assurance: bool
+        :param authentication_assurance_timeout: Number of seconds within which someone
+            must have authenticated to satisfy the policy
+        :type authentication_assurance_timeout: int
+        :param display_name: A user-friendly name for the policy
+        :type display_name: str
+        :param description: A user-friendly description to explain the purpose of the
+            policy
+        :type description: str
+        :param domain_constraints_include: A list of domains that can satisfy the policy
+        :type domain_constraints_include: iterable of str
+        :param domain_constraints_exclude: A list of domains that can not satisfy the
+            policy
+        :type domain_constraints_exclude: iterable of str
+
+        .. tab-set::
+
+            .. tab-item:: Example Usage
+
+                .. code-block:: pycon
+
+                    >>> ac = globus_sdk.AuthClient(...)
+                    >>> client_id = ...
+                    >>> r = ac.create_policy(
+                    ...     project_id="da84e531-1afb-43cb-8c87-135ab580516a",
+                    ...     high_assurance=True,
+                    ...     authentication_assurance_timeout=35,
+                    ...     display_name="No RedLight domain Policy",
+                    ...     description="Disallow access from @redlight.org",
+                    ...     domain_constraints_exclude=["redlight.org"],
+                    ... )
+                    >>> policy_id = r["policy"]["id"]
+
+            .. tab-item:: Example Response Data
+
+                .. expandtestfixture:: auth.create_policy
+
+            .. tab-item:: API Info
+
+                ``POST /v2/api/policies``
+
+                .. extdoclink:: Create Policy
+                    :ref: auth/reference/#create_policy
+        """
+        body: dict[str, t.Any] = {
+            "project_id": str(project_id),
+            "high_assurance": high_assurance,
+            "authentication_assurance_timeout": authentication_assurance_timeout,
+            "display_name": display_name,
+            "description": description,
+        }
+
+        if domain_constraints_include is not None:
+            body["domain_constraints_include"] = list(domain_constraints_include)
+        if domain_constraints_exclude is not None:
+            body["domain_constraints_exclude"] = list(domain_constraints_exclude)
+        return self.post("/v2/api/policies", data={"policy": body})
+
+    # TODO: Some API calls accept 'None' (null) so need to distinguish in update calls
+    # from when the caller did not provide a value.
+    # - update_policy():
+    #   - domain_constraints_include=[] != domain_constriants_include=None
+    # - update_client():
+    #   - terms_and_conditions=None
+    #   - privacy_policy=None
+    #   - required_idp=None
+    #   - preselect_idp=None
+    class _DefaultNull:
+        ...
+
+    _DEFAULT_NULL = _DefaultNull()
```

Comment:
I actually have a solution I want to put into the SDK for this problem of "null vs missing in python", which we need elsewhere as well. We're going to add a `globus_sdk.MISSING` sentinel value, which can be used just like this sentinel.

I'm going to ask that we move to using that -- since that means that it blocks this PR, I'll reprioritize it for later today.

---

Aside: my plan for globus-sdk v4 includes making all defaults use `MISSING`, never `None`, so that _any_ explicit call with `foo=None` can be interpreted as `"foo": null` properly, anywhere in the SDK.